app/src/main/java/io/apicurio/registry/rules/RulesServiceImpl.java

```diff
@@ -21,24 +22,74 @@
     @Inject
     @Current
     RegistryStorage storage;
-
-    public void validate(RuleContext context) {
-        switch (context.getRuleType()) {
-            case COMPATIBILITY:
-                if (!isCompatible(context.getArtifactType(), context.getLevel(), context.getArtifactId(), context.getContent())) {
-                    throw new RulesException(String.format("Incompatible artifact: %s [%s]", context.getArtifactId(), context.getArtifactType()));
-                }
-                break;
-            case SYNTAX_VALIDATION:
-                break;
-            case SEMANTIC_VALIDATION:
-                break;
+    
+    @Inject
+    RuleExecutorFactory factory;
+    
+    /**
+     * @see io.apicurio.registry.rules.RulesService#applyRules(java.lang.String, io.apicurio.registry.types.ArtifactType, java.lang.String, io.apicurio.registry.rules.RuleApplicationType)
+     */
+    @Override
+    public void applyRules(String artifactId, ArtifactType artifactType, String artifactContent,
+            RuleApplicationType ruleApplicationType) throws RuleViolationException {
+        boolean useGlobalRules = false;
+        @SuppressWarnings("unchecked")
+        List<String> rules = Collections.EMPTY_LIST;
+        if (ruleApplicationType == RuleApplicationType.update) {
+            rules = storage.getArtifactRules(artifactId);
+        }
+        if (rules.isEmpty()) {
```

Comment:
This would mean that the global rules are only applied if there is no artifact rule. Is this correct? Shouldn't the concept here be more like 'inheritance'? I'd suggest to describe the design in the `RuleService` interface.